astropy/io/fits/tests/test_convenience.py

```diff
@@ -153,3 +153,23 @@ def test_tabledump(self):
         # test with datafile
         fits.tabledump(filename, datafile=self.temp('test_tb.txt'))
         assert os.path.isfile(self.temp('test_tb.txt'))
+
+        # test with cdfile, hfile
+        """
+        test for https://github.com//astropy/astropy/issues/7015
+        """
+        fits.tabledump(filename, datafile = self.temp('test_tb.txt'), hfile = self.temp('hfile.txt'), cdfile = self.temp('cdfile.txt'), overwrite = True )
+        assert os.path.isfile(self.temp('hfile.txt'))
+        assert os.path.isfile(self.temp('cdfile.txt'))
+        sizeh =  os.path.getsize(self.temp('hfile.txt'))
+        sizecd =  os.path.getsize(self.temp('cdfile.txt'))
+        assert sizeh > 0
+        assert sizecd > 0
+        fits.tableload(self.temp('test_tb.txt'), hfile=self.temp('hfile.txt'), cdfile = self.temp('cdfile.txt'))
+        assert os.path.isfile(self.temp('test_tb.txt'))
+        os.remove(self.temp('test_tb.txt'))
+        os.remove(self.temp('hfile.txt'))
+        os.remove(self.temp('cdfile.txt'))
```

Comment:
Removing temporary files is not needed since it goes in a temp directory. Also you should remove the blank lines as they are causing the pep8 build failure.

Reply:
Hi @saimn I did all the suggestion above, and I created a new test file called test_tabledump.py,  I did rebase, commit, and test all things with py.test, no error.